app/src/test/java/org/simple/clinic/registration/register/RegistrationLoadingScreenControllerTest.kt

```diff
@@ -21,61 +19,84 @@ import org.simple.clinic.user.registeruser.RegistrationResult.UnexpectedError
 import org.simple.clinic.widgets.ScreenCreated
 import org.simple.clinic.widgets.UiEvent
 
-@RunWith(JUnitParamsRunner::class)
 class RegistrationLoadingScreenControllerTest {
 
   private val userSession = mock<UserSession>()
   private val screen = mock<RegistrationLoadingScreen>()
-  lateinit var controller: RegistrationLoadingScreenController
-
   private val uiEvents = PublishSubject.create<UiEvent>()
 
+  private val controller = RegistrationLoadingScreenController(userSession)
+
   @Before
   fun setUp() {
-    controller = RegistrationLoadingScreenController(userSession)
-
     uiEvents
         .compose(controller)
         .subscribe { uiChange -> uiChange(screen) }
   }
 
-  @Parameters(method = "params for testing user registration")
   @Test
-  fun `when screen is created then user registration api should be called`(results: RegistrationResult) {
-    whenever(userSession.register()).doReturn(Single.just(results))
-    whenever(userSession.clearOngoingRegistrationEntry()).doReturn(Completable.complete())
+  fun `when retry button is clicked, then loader should show and register api should be called`() {
+    whenever(userSession.register()).doReturn(Single.just<RegistrationResult>(Success))
+    whenever(userSession.clearOngoingRegistrationEntry()).doReturn(Completable.never())
```

Comment:
Curious, why a `Completable.never()` as opposed to a `Completable.complete()`?

Reply:
Let me recheck this. I don't exactly remember why I made this a `Completable.never`.

Reply:
The way the implementation is designed, clearing of ongoing entry does not impact going to the home screen after registration happens.

I had initially deleted the `Completable.complete` stub, but this broke the test because it needed a `Completable` to be returned from the `UserSession#clearOngoingRegistrationEntry` method.

I'll change this back to `Completable.complete` since it doesn't actually affect anything.